### ISSUE TITLE
Benchmarks with TSP instances

### ISSUE THREAD

**Issue by leonlan** [PyVRP/PyVRP#577]
#### Is your feature request related to a problem? Please describe
PyVRP is a VRP solver, but it is not unlikely that users also want to use PyVRP for TSPs. Currently, we don't have any benchmarks on TSP instances or any ideas on how well PyVRP performs on TSP instances.

#### Describe the solution you'd like
I want to know how well PyVRP performs on various TSP instances.
We can compare against [LKH-3](http://webhotel4.ruc.dk/~keld/research/LKH-3/), and also use those instances.

#### Additional context

> Bock, S., Bomsdorf, S., Boysen, N., & Schneider, M. (2024). A survey on the Traveling Salesman Problem and its variants in a warehousing context. European Journal of Operational Research. https://doi.org/10.1016/j.ejor.2024.04.014

This recent survey on TSPs in warehousing published in EJOR describes 11 variants, many of which we can already solve with PyVRP.

- [x] The classical TSP and its application to warehousing
- [ ] TSP with precedence constraints
- [ ] Clustered TSP
- [x] Generalized TSP
- [x] TSP with backhauls
- [x] Prize-collecting TSP
- [x] Orienteering problem
- [ ] Traveling Repairman Problem
- [ ] k-best TSP
- [x] TSP with time windows
- [ ] Covering Salesman Problem

**Comment by leonlan** [comment]
I don't expect PyVRP to scale very well to TSPs with >10K nodes (and it's not intended to be). One important reason is that `SegmentBetween` requires $O(n)$ merges and is used a lot in within-route moves, which is the only type of move used in TSPs.

Having some benchmarks on TSP is still interesting, just to get an idea of PyVRPs performance.

- TSPLIB
- LKH3: http://webhotel4.ruc.dk/~keld/research/LKH-3/
- GTSP: https://people.cs.nott.ac.uk/pszdk/gtsp.html#:~:text=In%20this%20section%20you%20can,of%20size%20N%20%E2%89%A4%201084

**Comment by N-Wouda** [comment]
For this I think we should have a benchmark on just barebones TSP. All the other aspects we already cover via VRP - the interesting bit here is only one vehicle, not the additional problem attributes.

Can we find instances of, say, 1-10K to test on?

**Comment by leonlan** [comment]
Yeah, we can take TSPLIB instances. There are about 20+ instances with more than 1K clients.
- https://github.com/mastqe/tsplib
- http://comopt.ifi.uni-heidelberg.de/software/TSPLIB95/STSP.html

They're all solved to optimality.

**Comment by leonlan** [comment]
We've got the instances (https://github.com/PyVRP/Instances/pull/14). Ran a local benchmark:

```
Instance  OK  Obj.        Iters. (#)  Time (s)  Gap (%)
--------  --  ----------  ----------  --------  -------
 pcb1173   Y     58247.0      199656   300.002     2.73
   d1291   Y     53112.0      189528   300.001     4.84
 nrw1379   Y     57412.0      152415   300.006     2.33
   u1432   Y    157272.0      170683     300.0     2.83
  rl1323   Y    282930.0      180065   300.002     4.78
  fl1577   Y     25209.0      157235   300.001     18.4
   u1817   Y     61061.0      125760   300.002     7.03
  rl1889   Y    340533.0      116918   300.001     7.68
   u2152   Y     68655.0       97583     300.0     7.13
   u2319   Y    236921.0       94564     300.0     1.14
  pr2392   Y    400021.0       80839   300.003     5.95
  fl3795   Y     38949.0       52946   300.002    49.26
 fnl4461   Y    196989.0       34450   300.004      8.8
  rl5915   Y    661221.0       28325   300.005    17.02
  rl5934   Y    639652.0       28215   300.001    15.16
 pla7397   Y  31094601.0       20010   300.009    33.68

     Avg. objective: 2148299
    Avg. iterations: 108074
      Avg. run-time: 300.00s
       Total not OK: 0
           Avg. gap: 11.80%
```

TSP is not our forte but I think that's fair. We're missing special TSP operators like 2OPT. I think it's more important to focus on large-scale VRP instances #735.
